lib/http.c

```diff
@@ -3993,9 +4037,19 @@ static CURLcode http_on_response(struct Curl_easy *data,
 #ifndef CURL_DISABLE_WEBSOCKETS
   /* All >=200 HTTP status codes are errors when wanting WebSocket */
   if(data->req.upgr101 == UPGR101_WS) {
-    failf(data, "Refused WebSocket upgrade: %d", k->httpcode);
-    result = CURLE_HTTP_RETURNED_ERROR;
-    goto out;
+    CURLcode r2;
+    if(!data->set.ws_upgrd_refused_ok) {
+      failf(data, "Refused WebSocket upgrade: %d", k->httpcode);
+      result = CURLE_HTTP_RETURNED_ERROR;
+      goto out;
+    }
+    infof(data, "WebSocket upgrade refused but continuing: %d", k->httpcode);
+    r2 = scheme_change_ws_to_http(data);
+    if(r2) {
+      result = r2;
+      infof(data, "Failed to switch from WebSocket to http scheme/handler");
+      goto out;
+    }
```

Comment:
```suggestion
    CURLcode result2;
    if(!data->set.ws_upgrd_refused_ok) {
      failf(data, "Refused WebSocket upgrade: %d", k->httpcode);
      result = CURLE_HTTP_RETURNED_ERROR;
      goto out;
    }
    infof(data, "WebSocket upgrade refused but continuing: %d", k->httpcode);
    result2 = scheme_change_ws_to_http(data);
    if(result2) {
      result = result2;
      infof(data, "Failed to switch from WebSocket to http scheme/handler");
      goto out;
    }
```
also to match a recent rename in master.